core/devops.go

```diff
@@ -209,7 +210,18 @@ func (d *Devops) invokeOrQuery(ctx context.Context, chaincodeInvocationSpec *pb.
 	}
 
 	// Now create the Transactions message and send to Peer.
-	uuid := util.GenerateUUID()
+	var customUUIDgenAlg = strings.ToLower(chaincodeInvocationSpec.UuidGenerationAlg)
+	var uuid string
+	var generr error
+	if customUUIDgenAlg != "" {
+		uuid, generr = util.GenerateUUIDWithAlg(customUUIDgenAlg, chaincodeInvocationSpec.ChaincodeSpec.CtorMsg.Args[0])
+		if generr != nil {
+			return nil, generr
+		}
+	} else {
+		uuid = util.GenerateUUID()
+	}
+	devopsLogger.Info("Transaction ID: %v", uuid)
```

Comment:
This assumes that the entire transaction is encoded base64 in the first parameter. We should split processing of CLI and rpc later, so rpc can be all binary.